src/broadcast_socket.cpp

```diff
@@ -56,10 +56,10 @@ using namespace std::placeholders;
 
 namespace libtorrent
 {
-	bool is_ip_address(char const* host)
+	bool is_ip_address(string_view host)
 	{
 		error_code ec;
-		address::from_string(host, ec);
+		address::from_string(host.data(), ec);
```

Comment:
there is an overload of ``from_string`` that takes a ``std::string const&`` though, so maybe the best solution for now is to make ``is_ip_address`` take that as its parameter